webflow-hacks/misc-hacks/6/6.ts

```diff
@@ -0,0 +1,69 @@
+document.addEventListener('DOMContentLoaded', function () {
+  const WAIT_TIME = 4500;
+  const COOKIE_NAME = 'seenGif';
+
+  const LOADING_WRAPPER_SELECTOR = '[fs-hacks-element="loading-wrapper"]';
+  const GIF_SELECTOR = '[fs-hacks-element="gif"]';
+  const CLEAR_BUTTON_SELECTOR = '[fs-hacks-element="clear-cookie"]';
+  const loadingWrapper = document.querySelector<HTMLDivElement>(LOADING_WRAPPER_SELECTOR);
+  const clearButton = document.querySelector<HTMLDivElement>(CLEAR_BUTTON_SELECTOR);
+  if (!loadingWrapper || !clearButton) return;
+
+  const seenGif = getCokie(COOKIE_NAME);
+  if (seenGif) return;
+
+  loadingWrapper.style.display = 'flex';
+
+  playGif(GIF_SELECTOR);
+  setCokie(COOKIE_NAME, 'true', 1);
+
+  setTimeout(async () => {
+    loadingWrapper.style.display = 'none';
+  }, WAIT_TIME);
+
+  clearButton.addEventListener('click', function () {
+    removeCookie(COOKIE_NAME);
+  });
+});
+
+/**
+ * define the playGif function
+ * @param {string} gifclass
+ **/
+function playGif(gifclass: string) {
+  // get the gif's class and store it in a variable
+  const gif = document.querySelector<HTMLImageElement>(gifclass);
+  if (!gif) return;
+  gif.src = gif.src.replace(/\?.*$/, '') + '?x=' + Math.random();
```

Comment:
I don't understand this line, would you mind explaining it?

Reply:
This reset the gif's src attribute by appending a random query string.
This will make the browser fetch the gif again every  when the page is reloaded, it will then play the gif

Reply:
I thought GIF images played on auto-repeat?

Reply:
This particular gif does not auto-repeat, I can set an infinite loop using an image editor, but that will mean code is unusable for people with single loop Gifs

Reply:
I see.
I would try to make sure people understand the purpose of this hack, because it has two underlying actions happening:
- Showing a loader based on a cookie. This could be any kind of loader (a lottie animation, a div that contains Webflow interactions, a video, an infinite-looping GIF...).
- Restarting a single loop GIF.

My first thought is to get rid of the GIF bit to avoid confusing users.